lib/core/src/server/preview/iframe-webpack.config.js

```diff
@@ -184,3 +183,18 @@ export default async ({
     },
   };
 };
+
+/**
+ * Return a string corresponding to template filled with bindings using following pattern:
+ * For each (key, value) of `bindings` replace, in template, `{{key}}` by escaped version of `value`
+ *
+ * @param template {String} Template with `{{binding}}`
+ * @param bindings {Object} key-value object use to fill the template, `{{key}}` will be replaced by `escaped(value)`
+ * @returns {String} Filled template
+ */
+const interpolate = (template, bindings) => {
```

Comment:
a third template exists in the manager `lib\core\src\server\manager\virtualModuleRef.template.js` as it's in the manager, I don't know if it's impacted by yarn 2 ? Also, not sure if there's any common place to put this helper to use at both places ? (if relevant)

Reply:
I didn't make a bigger refactoring because I have in mind that a TS migration + refactoring of `@storybook/core` is currently in progress in https://github.com/storybookjs/storybook/pull/9870. I'm not sure about the current state of this PR or even if it will be merged as it (for now there are a lot of conflicts). @ndelangen @shilman @kroeder do you know if it's going to land in 6.0? 

@tooppaaa there are also template things in `lib/core/src/server/utils/template.js`, it's for related to HTML file but it will be great to find a way to have only one "template mechanism". However, I think this is a refactoring that can be done outside this PR that fixes incompatibility between `preview.js` file and Yarn 2. @shilman are you ok too to merge this as it and work on another template related PR ?

Reply:
I'm fine with this landing in 6.0.0, what conflicts are you worries about?

Reply:
Any big change in `@storybook/core` will create conflicts in https://github.com/storybookjs/storybook/pull/9870 that why I would prefer to merge this one and do some refactoring after TS migration will be done.

Reply:
@gaetanmaisse hopefully the learnings from #9870 will be applicable to the refactor.

I'd agree merging this first rather then wait for #9870